browser/brave_stats/brave_stats_updater_browsertest.cc

```diff
@@ -321,45 +238,41 @@ IN_PROC_BROWSER_TEST_F(BraveStatsUpdaterBrowserTest, StatsUpdaterMigration) {
   EXPECT_STREQ(query_value.c_str(), "BRV001");
 }
 
+class BraveStatsUpdaterReferralCodeBrowserTest
+    : public BraveStatsUpdaterBrowserTest {
+ public:
+  void SetUp() override {
+    ASSERT_TRUE(dir.CreateUniqueTempDir());
+    const base::FilePath promo_code_file =
+        dir.GetPath().AppendASCII("promoCode");
+    WritePromoCodeFile(promo_code_file, referral_code());
+    brave::BraveReferralsService::SetPromoFilePathForTesting(promo_code_file);
+    BraveStatsUpdaterBrowserTest::SetUp();
+  }
+
+  int WritePromoCodeFile(const base::FilePath& promo_code_file,
+                         const std::string& referral_code) {
+    base::ScopedAllowBlockingForTesting allow_blocking;
+    return base::WriteFile(promo_code_file, referral_code.c_str(),
```

Comment:
WriteFile returns a bool so you can put an ASSERT on it.

Reply:
I know this is pre-existing code, but make the most of this change to also write `base::WriteFile(promo_code_file, referral_code)` here and avoid the std::string -> C string conversion.